### DISCUSSION TITLE
Will it cause any compatibility issues for billboard.js when C3 and D3 are also included in to global scope?

### DISCUSSION THREAD

**Discussion by 5r1m** [naver/billboard.js#2301] [Q&A] [answered]
Is it possible to use C3 and billboard.js (with pkg as it includes d3) together? Reason is we have some old charts which are built with C3 and D3 and we are moving towards billboard.js how ever for some duration we will have to include both of them, meaning C3, D3 (old version compatible with C3) with exposing them to global scope and then new packaged billboard.js libraries. With the older D3 in global scope will it cause any problem for billboard.js?

Does billboard.js support providing alias D3?

**Comment by netil** [comment]
Hi @5r1m, basically c3 & billboar.js uses different global namespace, so in that sense I guess will not have issue.
But as you know, C3 will support `d3 v4 ~ v6`, where the latest billboard v3+ has `d3 v7` dependency.

Is not difficult to test on this, so try.

**Comment by 5r1m** [comment]
Thanks @netil . I have already tried this and it works. I also asked this for few other reasons:
- In "https://naver.github.io/billboard.js/release/latest/doc/", Its mentioned as "Packaged version is not an official distribution". It was not clear for me on what is this means? Will it have changes in future like stopping support etc or its just to mention that when used we need to explicitly use packaged version to be loaded instead of non-packaged version?
- Was also concerned where some modules pollute globals (its against principle, but have seen them earlier :( ) and wanted to make sure that it stays this way when using packaged version
- In "https://github.com/naver/billboard.js/wiki/How-to-bundle-for-legacy-browsers%3F" its mentioned as "you need to add on your bundler configuration to not concatenate d3 modules". This I assume is to avoid duplicity only

All this made me to be ask this question so that it doesnt create problem in future if we take this path. 

Thank you once again for this library and your support.

**Comment by netil** [comment, marked as answer]
> Packaged version is not an official distribution

The "packaged" means, the distribution comes with the d3.js. We mentioned to be not as an 'official', because in general billboard.js will be consumed as module. Assuming each project has its own build configurations, where in most cases project owners want to control how to bundle it, is not adequate provide dependent module(d3.js) bundled as part of billboard.js package.

So, based on that criteria, "packaged" distribution is only to facilitate those don't have own bundling necessity or being consumed as traditional script loading.

Even described as "unofficial", the packaged build will be provided for future releases.

> global pollution

if you load the libraries(not only for billboard.js) as UMD, most libraries will expose their namespace as global, because is the only way to provide an access point. In case of billboard.js, will expose "window.bb" only as global namespace.

> "you need to add on your bundler configuration to not concatenate d3 modules"

If you opt to use "packaged" build, don't have to. But if you're using the library importing as module(ESM/CJS), you need to handle d3.js transpiling for legacy browsers.

**Reply by 5r1m** [reply]
Thanks @netil for taking time to clarify these questions
